0000-stream/0000-stream.md

```diff
@@ -0,0 +1,174 @@
+
+Multiple Money Streams MAY be used to separately account for value sent and received for different purposes. A Money Stream can be long-lived and the send and receive limits MAY be adjusted over time (for example, in response to an application-level action) or a Money Stream can be used to account for a single logical payment, in which case it would be closed when the payment is finished.
+
+Money Streams are identified by integers starting from 1. Odd-numbered Money Streams are initiated by the Client and even-numbered Money Streams are initiated by the Server. Money Stream identifiers are unrelated to Data Stream identifiers and a single Connection MAY have Money Streams and Data Streams with the same number.
+
+### Multiplexed Data Streams
+
+STREAM also offers the ability to send data through the Connection over ILP (rather than over the Internet). Data Streams are the abstraction used for grouping bytes sent over the connection. Data Streams are heavily inspired by [QUIC Bidirectional Streams](https://tools.ietf.org/html/draft-ietf-quic-transport-10#section-10).
+
+Data Streams are identified by integers starting from 1. Odd-numbered Data Streams are initiated by the Client and even-numbered Data Streams are initiated by the Server. Data Stream identifiers are unrelated to Money Stream identifiers and a single Connection MAY have Data Streams and Money Streams with the same number.
+
+### Stream and Connection Flow Control
+
+STREAM implements stream- and connection-level flow control. Clients and Servers communicate the maximum amount of money or data they are willing to receive on a given stream or for the connection as a whole. Limits may be changed and retransmitted throughout the life of the Connection to accept more money or data as desired by the application.
+
+### Connection Migration
+
+Clients and Servers MAY change the address the other entity will send packets to at any time by sending a `ConnectionNewAddress` frame with their new ILP address.
+
+**TODO**: Should we stipulate that the new ILP address is for an account in the same currency or a completely different one? (Should we forget the exchange rate we computed before if the address changes?)
+
+### Authenticated Encryption
+
+All STREAM packets are encrypted using [AES-256-GCM](https://en.wikipedia.org/wiki/Galois/Counter_Mode) with a 12-byte Initialization Vector (IV) and a 16-Byte Authentication Tag.
+
+If subsequent versions support additional encryption algorithms, those details should be exchanged between the sender and receiver when they establish the shared secret. If a receiver attempts to decrypt an incoming packet but is unable to (perhaps because the sender is using an unsupported cipher), they SHOULD reject the incoming transfer with an `F06: Unexpected Payment` error.
```

Comment:
Is there any difference between a corrupted message and an unsupported cipher?

Reply:
Nope